Fix one piece of zen-parse@gmx.net's advisory on cookie files.  Modified from a
previous version to only log an error if setgroups() fails AND we're the
superuser.

--- openssh-3.6.1p2/sshd.c	Sun Apr 15 22:00:02 2001
+++ openssh-3.6.1p2/sshd.c	Thu Jun  7 14:50:57 2001
@@ -757,6 +757,17 @@
 #ifdef HAVE_SCO_PROTECTED_PW
 	(void) set_auth_parameters(ac, av);
 #endif
+	/* Clear out any supplemental groups we may have inherited.  This
+	 * prevents inadvertent creation of files with bad modes (in the
+	 * portable version at least, it's certainly possible for a pluggable
+	 * authentication module to create a file, and we can't control the
+	 * code in every module which might be used). */
+	if ((setgroups(0, NULL) == -1) && (geteuid() == 0)) {
+		if (debug_flag)
+			error("setgroups() failed: %.200s", strerror(errno));
+		else
+			fatal("setgroups() failed: %.200s", strerror(errno));
+	}
 
 	/* Initialize the log (it is reinitialized below in case we forked). */
 	if (debug_flag && !inetd_flag)
